Added following feature: ESLint

Daniel0K opened the pull request:
EsLint configured for ts, spec.ts and html

Daniel0K committed:
Added Lint to project

Daniel0K committed:
Исправлена ошикба в тестах, после рефакторинга метода

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -18,39 +18,35 @@ export class AppComponent {
     TestActionsService.setActivePin(TestActionsService.houses[0]);
-    if (document.cookie == '') {
-    } else {
+    if (document.cookie !== '') {
```
Куки чаще всего не пустые, т.к. в них всегда может лежать что то, так что такая проверка тебе точно не нужна

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -18,39 +18,35 @@ export class AppComponent {
     }
   }
 
+  hasCookie(value: string): boolean {
```
Такой метод надо класть в сервис для работы с куками, так как использоваться он может не только в этой компоненте
Плюс есть такой принцип как "Принцип разделения ответственности", буква S в SOLID

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -18,39 +18,35 @@ export class AppComponent {
-        this.TestActionsService.houses.push(this.cookiesTest[i]);
-      }
+      this.cookiesTest.forEach((item) => {
+        this.TestActionsService.houses.push(item);
```
Ты просто добавляет элементы одного массива в другой, для такого есть метод concat

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -18,39 +18,35 @@ export class AppComponent {
-          this.cookiesTest[i];
-      }
+      this.cookiesTest.forEach((item) => {
+        this.TestActionsService.houses[item.id] = item;
```
Искать элемент лучше через метод find и не пользоваться обращением по элементу массива

Daniel0K replied:
this.TestActionsService.houses[this.TestActionsService.houses.findIndex(el => el.id === item.id)] =item;
Так чтоль?

YaroslavTrubanov commented on src/app/pins-form/pins-form.component.ts:
```diff
@@ -31,16 +31,15 @@ export class PinsFormComponent implements DoCheck {
     };
     if (
-      this.TestActionsService.checkCrossings4DatesPeriod(
+      !this.TestActionsService.checkCrossings4DatesPeriod(
```
Лучше переименовать метод, так как сам метод говорить "Проверить пересечение...", что подразумевает его использование как функции без возвращаемого значения, а ты используешь его в качестве проверки в конструкции if else
Например, подойдёт: "isBookingExists" если я правильно понял смысл метода. Ну вообщем направление ты понял)

YaroslavTrubanov commented on src/app/services/test-actions.service.ts:
```diff
@@ -106,18 +106,18 @@ export class TestActionsService {
       new RegExp(
         '(?:^|; )' +
-          name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1') +
+          name.replace(/([\.$?*|{}\(\)\[\]\\\/\+^])/g, '\\$1') + //eslint-disable-line
```
Неа, не пойдет)

YaroslavTrubanov commented on src/app/services/test-actions.service.ts:
```diff
@@ -10,16 +15,10 @@ export interface Pin {
 @Injectable({
   providedIn: 'root',
 })
 export class TestActionsService {
```
Название сервиса странное, непонятно что делает
Также здесь работа и с куками и с бронированием, надо разделять
Почитай SOLID

Daniel0K committed:
Поправлены замечания в MR#7
Разнесены функции по соответствующим классам согласно солид.
Изменено имя метода.

YaroslavTrubanov commented on src/app/models/pin-model.ts:
```diff
@@ -0,0 +1,11 @@
+import {bookings} from "./bookings-model";
+
+export interface pin {
```
название интерфейса то все равно с большой буквы надо писать, а вот имя файла с маленькой, и model можно там не писать - просто pin.ts

YaroslavTrubanov commented on src/app/models/bookings-model.ts:
```diff
@@ -0,0 +1,4 @@
+export interface bookings {
```
с большой надо

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -10,58 +12,49 @@ export class AppComponent {
     }
   }
 
   checkUpdatedPinCookies() {
```
метод че то проверяет по название, а внутри происходят какие то присваивания

YaroslavTrubanov commented on src/app/app.component.ts:
```diff
@@ -10,58 +12,49 @@ export class AppComponent {
-      this.cookiesTest = <Pin[]>(
-        JSON.parse(this.TestActionsService.getCookie('updatedPins') || '')
+    if (this.cookiesService.hasCookie('updatedPins')) {
+      this.cookiesTest = <pin[]>(
```
че то странное название переменной

Daniel0K committed:
Исправлены замечания по MR#7 v3
Исправлены названия методов и переменных
Изменена капитализация интерфейсов

Daniel0K committed:
Исправлены замечания по MR#7 v4
Не до конца поправил изменения имени интефейса. не во всех файлах

Daniel0K committed:
Исправлены замечания по MR#7 v5
Упал билд:не до конца поправил изменения имени интефейса. не во всех файлах.
